hal/device_uid_api.h

```diff
@@ -0,0 +1,73 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DEVICE_UID_API_H
+#define MBED_DEVICE_UID_API_H
+
+#include <stddef.h>
+#include "device.h"
```

Comment:
is this inclusion needed? `device.h` was replaced by `device_has` macros and config, might be cleaned up soon.

I noticed this header file is included in each hal implementation. I would expect it includes only deviceuid hal header file, or anything else that is needed there.

Reply:
to be honest: I just took this from the other hal implementations ;)